### DISCUSSION TITLE
Jobs for Operating Guns disappear when certain periscopes are routed through circuit boxes (Tested with Orca, happens with 3rd and 4th periscope counting from the left)

### DISCUSSION THREAD

**Discussion by DgtUC** [FakeFishGames/Barotrauma#13593] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

The initial problem was explained in this reddit post:
https://www.reddit.com/r/Barotrauma/comments/1bj8jsh/troubleshooting_periscope_control_switch_on_the/

So far I've only encountered this issue with the Orca. I am quite new to the game and haven't tried to see if the issue is replicated on other subs, or even custom subs (because I'm too much of a newbie to build one)

After testing it some more, I found out that if the connection between periscopes #3 and #4 (counting from the left) are routed through a circuit box and there is any modules between the inputs and outputs, the jobs for operating guns disappear for those periscopes. 

To doublecheck this, I wired all the periscopes to their respective guns and hull camera/spotlight, but made the wires go through a circuit box. If the periscopes are wired to the guns (and hull camera) through the circuit box without any modules in between, it all works fine. Putting any sort of logic circuits in between the inputs and outputs of the circuit box for the first two periscopes will work just fine, as well, and the AI bots are still able to be assigned to operating guns. 

However, adding any sort of logic circuitry between the 3rd and 4th periscope makes the jobs disappear. Furthermore, I've determined this seems to be an issue with the circuit box itself, because bringing those logic circuits outside of the circuit box and assembling them with plain components onto the walls doesn't cause this bug - I'm still able to use my turret switcher and the jobs don't disappear. It's just abysmal to look at.

Please fix, I was loving the circuit box.

### Reproduction steps

This is assuming you are running the Orca. 
1. Route periscope connections to guns via a circuit box
2. Connect all cables directly - no jobs should disappear.
3. Place any sort of logic modules in between the first 2 periscope's wires (e.g. a round component) - works fine.
4. Do the same for the last two periscopes and the jobs disappear. 

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.2.8.0 (Winter Update hotfix 2)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
Haven't tried it on Linux yet, but I doubt it's platform dependent.
```

**Comment by Regalis11** [comment]
This seems like it's probably caused by the same issue as https://github.com/FakeFishGames/Barotrauma/issues/13343, should be fixed in the latest unstable build. and the fix will be included in the next update. Sorry for the inconvenience!
